contracts/modules/cover/Cover.sol

```diff
@@ -402,6 +416,21 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     emit CoverEdited(coverId, cover.productId, lastCoverSegmentIndex + 1, msg.sender);
   }
 
+  function checkFixedPricingAllowedPools(uint productId, PoolAllocationRequest[] memory allocationRequests) internal {
+
+    uint[] memory allowedPools = fixedPricingAllowedPools[productId];
```

Comment:
The reason i made this an array and not a mapping is that because when i do setProducts i want to overwrite it completely.

You can't do that with a mapping, because you can't wipe a mapping clean with 1 instruction